src/core/ddsc/src/dds_stream.c

```diff
@@ -710,7 +710,7 @@ static void dds_stream_read (dds_istream_t * __restrict is, char * __restrict da
           case DDS_OP_VAL_4BY: *((uint32_t *) addr) = dds_is_get4 (is); ops += 2; break;
           case DDS_OP_VAL_8BY: *((uint64_t *) addr) = dds_is_get8 (is); ops += 2; break;
           case DDS_OP_VAL_STR: *((char **) addr) = dds_stream_reuse_string (is, *((char **) addr), 0); ops += 2; break;
-          case DDS_OP_VAL_BST: dds_stream_reuse_string (is, (char *) addr, ops[2]); ops += 3; break;
+          case DDS_OP_VAL_BST: addr = (void *) dds_stream_reuse_string (is, (char *) addr, ops[2]); ops += 3; break;
```

Comment:
That won't do: it introduces a dead store. (I am surprised Coverity doesn't complain about it!)

Reply:
Split the function into two different functions: one for string (returning the string) and one for bounded string (not returning anything).